src/prop-types/childrenOf.js

```diff
@@ -21,7 +22,13 @@ const childrenOf = expectedChildTypes => {
         return;
       }
       const childDisplayName = getDisplayName(child.type || child);
-      if (!expectedChildTypes.includes(child.type)) {
+      const expectedChildType = expectedChildTypes.find(
+        c =>
+          typeof c === 'string'
+            ? c === childDisplayName
+            : c.name === childDisplayName
+      );
+      if (!areComponentsEqual(child.type, expectedChildType)) {
         throw new Error(
```

Comment:
@joshblack Thinking more about this issue - Should the condition here *really* be considered fatal?

Reply:
Good point! Should be returning the error instead of throwing it 🙂 Must have missed this in an earlier PR.